### DISCUSSION TITLE
When the production branch has changed then deploy latest commit to production

### DISCUSSION THREAD

**Discussion by sztadii** [vercel/vercel#8583] [Ideas]
With a team, we will use a trunk-based development pattern in our project.
I was going to change a production branch and I was expecting that it will take the latest commit from that branch and deploy it.

Unfortunately, I need to find that commit, promote it to production ( hopefully I will find a proper commit ) and when I will make a quick fix then it will redeploy automatically.

Can you improve it?

**Comment by sztadii** [comment]
It would be nice to have a button "Promote branch to production" or something like that.

**Comment by amyegan** [comment]
Hi @sztadii! It sounds like you are using [Vercel for GitHub](https://vercel.com/docs/concepts/git/vercel-for-github) to automatically deploy your code. Did you already change the Production Branch in your Project Settings?
<img width="810" alt="image" src="https://user-images.githubusercontent.com/8247105/191098613-f9a13a73-45f8-4b1e-915b-3f7340c51e47.png">

Saving that setting will not have an immediate effect on your current production deployment. But all commits to the branch _after_ that setting is saved will automatically be deployed to production for you unless you have modified the behavior in `vercel.json` or with an Ignored Build Step script. 

Your existing preview deployments should have the option to "Promote to Production" from your Deployments list.
<img width="782" alt="image" src="https://user-images.githubusercontent.com/8247105/191099778-2b080ec1-48df-404d-ae69-e5cd0a5687a8.png">

And you can filter the list by branch to make deployments easier to find.
<img width="521" alt="image" src="https://user-images.githubusercontent.com/8247105/191100035-95ec6f39-8739-41a7-a3e6-dd2229632fc9.png">
